Change Tanner traits

maidxn opened the pull request:
1. In first 7 days, if Tanner doesn't vote anyone in the morning, Tanner will die
2. After 7 days, the rule in 1 is not applied anymore and Tanner will become a normal villager

huybt22 commented:
Please also have a look at these parts:

https://github.com/dnh-bot/dnh-werewolf-bot/blob/main/json/role_info.json#L134
https://github.com/dnh-bot/dnh-werewolf-bot/blob/main/game/text_template.py#L209

huybt22 commented on game/__init__.py:
```diff
@@ -1235,6 +1260,24 @@ async def get_hunted_target_on_hunter_death(self, hunter):
                     return hunted
         return None
 
+    def get_tanner_id(self):
```
you may consider this:

def get_alive_tanner_id(self):
    tanner_id = next((player.player_id for player in self.players.value() if player.is_alive() and isinstance(player, roles.Tanner)), None)
    return tanner_id

neihousaigaai replied:
Following to @huybt22 suggested code, should we need a more generic function here? It should be used many other places as well.

```python
def get_alive_players_with_roles(self, roles=None):
    _id = next((player.player_id for player in self.players.value() if player.is_alive() and (roles is None or isinstance(player, roles)), None)
    return _id
```

maidxn replied:
I'll update it

maidxn commented:
- I updated the attributes in Tanner class to remove some variables need to track in main flow.
- I also changed Tanner party to Volatile Allegiance [ref](https://werewolves.es/volatile-allegiance/) here, I think when we use this party, it may help us in the future for some roles that can change party in the game.

nitro2 reviewed (dismissed):
I think it's ok for now. You can merge it.
